linetools/spectralline.py

```diff
@@ -0,0 +1,370 @@
+        ''' Fill atomic data and setup analy
+        Parameters:
+        -----------
+        linelist : LineList, optional
+          Class of linelist or str setting LineList
+        '''
+
+        # Deal with LineList
+        if linelist is None:
+            self.llist = LineList('ISM')
+        elif isinstance(linelist,basestring):
+            self.llist = LineList(linelist)
+        elif isinstance(linelist,LineList):
+            self.llist = linelist
+        else:
+            raise ValueError('Bad input for linelist')
+
+        # Data
+        self.data.update(self.llist[self.wrest])
+
+        #
+        self.analy.update( {
+            'FLG_EYE': 0,
+            'FLG_LIMIT': 0, # No limit
+            'DATFIL': '', 
+            'IONNM': self.data['name']
+            })
+
+        # Additional attributes for Absorption Line
+        self.attrib.update({'N': 0., 'Nsig': 0., 'flgN': 0, # Column
+                       'b': 0., 'bsig': 0.  # Doppler
+                       } )
+
+    # Perform AODM on the line
+    def aodm(self, **kwargs): 
+        """  AODM calculation
+
+        Parameters
+        ----------
+        spec : Spectrum1D (None)
+          1D spectrum.  Required but often read in through the Class (self.spec)
+        conti : np.array (None)
+          Continuum array 
+
+        Returns:
+          N, sigN : Column and error in linear space
+        """
+
+        # Grab Spectrum
+        spec = self.set_spec(**kwargs)
+
+        # Velocity array
+        spec.velo = spec.relative_vel(self.wrest*(1+self.analy['z']))
+
+        # Pixels for evaluation
+        pix = spec.pix_minmax(self.analy['z'], self.wrest,
+                        self.analy['VLIM'].to('km/s'))[0]
+                        #self.analy['VLIM'].to('km/s').value)[0]
+
+        # For convenience + normalize
+        velo = spec.velo[pix]
+        fx, sig = parse_spec(spec, **kwargs)
+
+        # dv
+        delv = velo - np.roll(velo,1)
+        delv[0] = delv[1]
+
+        # Atomic data
+        cst = (10.**14.5761)/(self.data['fval']*self.wrest) / (u.km/u.s) / u.cm * (u.AA/u.cm)
+
+        # Mask
+        mask = (pix == pix) # True = good
+        nndt = Quantity(np.zeros(len(pix)), unit='s/(km cm cm)')
+
+        # Saturated?
+        satp = np.where( (fx <= sig/5.) | (fx < 0.05) )[0]
+        if len(satp) > 0:
+            mask[satp] = False
+            lim = np.where(sig[satp] > 0.)[0]
+            if len(lim) > 0:
+                sub = np.maximum(0.05, sig[satp[lim]]/5.)
+                nndt[satp[lim]] = np.log(1./sub)*cst
+                flg_sat = len(lim)
+        # AODM
+        nndt[mask] = np.log(1./fx[mask])*cst
+
+        # Sum it
+        ntot = np.sum( nndt*delv )
+        tvar = np.sum( (delv*cst*sig/fx)**2 )
+
+        # Fill
+        self.attrib['N'] = ntot
+        self.attrib['sigN'] = np.sqrt(tvar)
+        logN, sig_logN = xsb.lin_to_log(self.attrib['N'].value, self.attrib['sigN'].value)
```

Comment:
Need to add lin_to_log to line_tools (we don't want xastropy as a requirement, right?)